proksee/species_estimator.py

```diff
@@ -161,27 +161,42 @@ def run_mash(self):
 
         LINE_LENGTH_LIMIT = 3500  # Actually 4095, but smaller here for safety.
 
-        output_filepath = os.path.join(self.output_directory, self.OUTPUT_FILENAME)
+        output_filepath = os.path.abspath(os.path.join(self.output_directory, self.OUTPUT_FILENAME))
+
+        # Find the common directory between all paths:
+        # This is a safer solution than assuming everything will always use the output directory.
+        common_path = os.path.commonpath(self.input_list)
+
+        if os.path.isdir(common_path):
+            common_directory = common_path
+        else:
+            # The common path is NOT a directory.
+            # This can happen when there is only a single input (the common path is the single file's filepath).
+            common_directory = os.path.dirname(common_path)
 
         # create the mash command
+        # Use the full file path for the database file:
         command = "mash screen -i 0 -v 1 " + self.mash_database_filename
 
         for item in self.input_list:
-            command += " " + str(item)
+            # Grab the relative path from the common directory of each item:
+            command += " " + str(os.path.relpath(item, start=common_directory))
 
-            # Break loop of command line argument is getting too long.
+            # Break loop if command line argument is getting too long.
             # This behaviour is likely fine for now, since the contigs are organized by size
             # and the missed contigs will likely be uninformative.
             if len(command) >= LINE_LENGTH_LIMIT:
-                continue
+                break
```

Comment:
You might want to consider logging a message on which contig you stopped at for estimating the species and how many contigs were used/how many were left? For example: **Species estimation performed using only 500/1000 of the longest contigs (excludes contig500 and shorter).**

Reply:
Added.